src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/ApplicationsController.cs

```diff
@@ -0,0 +1,326 @@
+            {
+                throw new ApplicationException("ApplicationId cannot be null or contain forward slash /");
+            }
+
+            string[] parts = applicationId.Split("-");
+
+            if (parts.Length > 1)
+            {
+                return parts[0];
+            }
+
+            throw new ApplicationException("Cannot get application Owner Id from applicationId: {applicationId}");
+        }
+
+        /// <summary>
+        /// Updates an instance
+        /// </summary>
+        /// <returns></returns>
+        [HttpPut("{applicationId}")]
+        public async Task<ActionResult> Put(string applicationId, [FromBody] ApplicationMetadata application)
+        {
+            string applicationOwnerId = GetApplicationOwner(applicationId);
+            ApplicationMetadata existingApplication;
+
+            try
+            {
+                existingApplication = await repository.FindOne(applicationId, application.ApplicationOwnerId);
+            }
+            catch (Exception e) 
+            {
+                return NotFound($"Unable to find application with applicationId={applicationId} for applicationOwnerId={applicationOwnerId} for update: {e.Message}");
+            }
+
+            if (application == null)
+            {
+                return BadRequest("Missing application metadata object. Please attach one.");
+            }
+
+            if (application.Id == null || !application.Id.Equals(applicationId))
+            {
+                return BadRequest("applicationId in path does not match id in attached object");
+            }
+
+            if (application.ApplicationOwnerId == null || !application.ApplicationOwnerId.Equals(applicationOwnerId))
+            {
+                return BadRequest("ApplicationOwnerId from applicationId is not matching attached object");
+            }
+
+            application.LastChangedBy = User.Identity.Name;
+            application.LastChangedDateTime = DateTime.UtcNow;
+
+            // Make sure client has not updated any important fields
+            application.CreatedBy = existingApplication.CreatedBy;
+            application.CreatedDateTime = existingApplication.CreatedDateTime;
+
+            try
+            {
+                ApplicationMetadata result = await repository.Update(application);                
+
+                return Ok(result);
+            }
+            catch (DocumentClientException dce)
+            {
+                if (dce.Error.Code.Equals("NotFound"))
+                {
+                    return NotFound($"Did not find application with id={applicationId} to update");
+                }
+
+                return StatusCode(500, $"Document database error: {dce.Message}");
+            }
+            catch (Exception e) 
+            {
+                return StatusCode(500, $"Unable to perform request: {e.Message}");
+            }
+        }
+
+        /// <summary>
+        /// Delete an instance
```

Comment:
Comments should be updated to match the code

Reply:
ok